test.py

```diff
@@ -20,6 +21,14 @@ def setUp(self):
     def tearDown(self):
         if 'cuda' in str(self):
             memory = torch.cuda.memory_allocated()
+            gc.collect()
+            # Deleting any tensors which were not picked up by GC
+            for obj in gc.get_objects():
+                try:
+                    if torch.is_tensor(obj) or (hasattr(obj, 'data') and torch.is_tensor(obj.data)):
+                        del obj
```

Comment:
this is actually a fallacy i think. 

gc.collect() should cause all 'deletable' objects to be deleted.  the ones that remain can't be deleted because someone holds references to them.  getting 'obj' from gc.get_objects() actually creates a new reference to the obj, so if it had 1 refcount before, now it has 2.  then del obj just sets it back to 1.  so, no benefit, i think.

i think adding gc.collect to master makes sense.  

the gc.get_objects part should be just a diagnostic though, for debugging.  to help you identify what holds the references to 'obj' that kept it from being collected.   not sure we should commit that part at all, or at least it should be separated out as a utility function or something...

Reply:
This makes sense. The del obj actually didn't do anything. I think this was my misunderstanding that it helped.  I removed this part.